src/commonMain/kotlin/fr/acinq/lightning/channel/Normal.kt

```diff
@@ -671,7 +687,10 @@ data class Normal(
     /** If we haven't completed the signing steps of an interactive-tx session, we will ask our peer to retransmit signatures for the corresponding transaction. */
     fun getUnsignedFundingTxId(): ByteVector32? = when (spliceStatus) {
         is SpliceStatus.WaitingForSigs -> spliceStatus.session.fundingTx.txId
-        else -> null
+        else -> when (commitments.latest.localFundingStatus.signedTx) {
+            null -> commitments.latest.localFundingStatus.txId
+            else -> null
+        }
```

Comment:
Nice catch again! I think rewriting the method like so is more explicit (same style as your refactoring for `CommitSig`):
```kotlin
fun getUnsignedFundingTxId(): ByteVector32? = when {
        spliceStatus is SpliceStatus.WaitingForSigs -> spliceStatus.session.fundingTx.txId
        commitments.latest.localFundingStatus is LocalFundingStatus.UnconfirmedFundingTx && commitments.latest.localFundingStatus.sharedTx is PartiallySignedSharedTransaction -> commitments.latest.localFundingStatus.txId
        else -> null
    }
```

Reply:
Agreed, done in https://github.com/ACINQ/lightning-kmp/pull/438/commits/9b9810ffc67da8f0b40825b2126d1dc54ee6b0dd